src/fusiontensor/array_cast.jl

```diff
@@ -0,0 +1,203 @@
+# This file defines interface to cast from and to generic array
+
+using BlockArrays: AbstractBlockArray, BlockedArray, blockedrange, blocklengths, findblock
+
+using BlockSparseArrays: BlockSparseArrays, BlockSparseArray
+using GradedUnitRanges: AbstractGradedUnitRange, blocklabels
+using SymmetrySectors: block_dimensions, quantum_dimension
+using TensorAlgebra: contract
+
+# =================================  High level interface  =================================
+
+#### cast from array to symmetric
+function FusionTensor(
+  array::AbstractArray,
+  codomain_legs::Tuple{Vararg{AbstractGradedUnitRange}},
+  domain_legs::Tuple{Vararg{AbstractGradedUnitRange}},
+)
+  return to_fusiontensor(array, codomain_legs, domain_legs)
+end
+
+#### cast from symmetric to array
+function BlockSparseArrays.BlockSparseArray(ft::FusionTensor)
+  return to_array(ft)
+end
+
+# =================================  Low level interface  ==================================
+function to_fusiontensor(
+  array::AbstractArray,
+  codomain_legs::Tuple{Vararg{AbstractGradedUnitRange}},
+  domain_legs::Tuple{Vararg{AbstractGradedUnitRange}},
+)
+  bounds = block_dimensions.((codomain_legs..., domain_legs...))
+  blockarray = BlockedArray(array, bounds...)
+  return to_fusiontensor(blockarray, codomain_legs, domain_legs)
+end
+
+get_tol(a::AbstractArray) = get_tol(real(eltype(a)))
+get_tol(T::Type{<:Integer}) = get_tol(Float64)
+get_tol(T::Type{<:Real}) = 10 * eps(T)
```

Comment:
A naming convention I have been using is defining functions that provide default values for a keyword argument `k` as `default_k(...)`, so here it would be `default_tol(...)`.

Also, here's a suggestion for making it a bit more compact and general:
```julia
default_tol(a::AbstractArray) = default_tol(eltype(a))
default_tol(arrayt::Type{<:AbstractArray}) = default_tol(eltype(arrayt))
default_tol(elt::Type{<:Number}) = 10 * eps(real(float(elt)))
```